rollup.config.js

```diff
@@ -65,16 +66,32 @@ const external = (id) =>
   !id.startsWith('\0') && !id.startsWith('.') && !id.startsWith('/');
 
 // Pragmatically create a Rollup config for each package
-const getRollupConfig = ({ pwd, babelConfigFile }) => {
+const getRollupConfig = ({ pwd, babelConfigFile, isModule }) => {
   const SOURCE_DIR = path.resolve(pwd);
+  const PRIVATE_PATH = `${SOURCE_DIR}/src/private.js`;
+
   // Get the package.json file
   const pkgConfig = require(`${SOURCE_DIR}/package.json`);
+
   // Relative input location for Rollup to bundle from
-  const input = [`${SOURCE_DIR}/src/index.js`, `${SOURCE_DIR}/src/private.js`];
+  const input = [`${SOURCE_DIR}/src/index.js`];
+
+  // Not all packages have private exports
+  if (fs.existsSync(PRIVATE_PATH)) {
+    input.push(PRIVATE_PATH);
+  }
 
-  // Shared Rollup plugins
+  // Shared Rollup plugins for code-only packages
   const rollupPlugins = [nodeResolve(), postcss(getPostCSSOptions()), json()];
 
+  // Gather info about the package
+  const IS_CJS = Boolean(pkgConfig?.main);
+  const IS_ESM = pkgConfig?.type === 'module' || Boolean(pkgConfig?.module);
+  const TYPE = pkgConfig?.type;
+
+  // check if only ESM is part of the output
+  const ESM_ONLY = IS_CJS && !IS_ESM && TYPE === 'module';
```

Comment:
This logic is still inconsistent. Will revise to align with the presence of "module" or not.